packages/fuels-core/src/code_gen/abigen.rs

```diff
@@ -195,12 +258,21 @@ impl Abigen {
     }
 
     pub fn functions(&self) -> Result<TokenStream, Error> {
-        let tokenized_functions = self
-            .abi
-            .functions
-            .iter()
-            .map(|function| expand_function(function, &self.types))
-            .collect::<Result<Vec<TokenStream>, Error>>()?;
+        let tokenized_functions = match self.is_script {
+            false => self
+                .abi
+                .functions
+                .iter()
+                .map(|function| expand_function(function, &self.types))
+                .collect::<Result<Vec<TokenStream>, Error>>()?,
+            true => self
+                .abi
+                .functions
+                .iter()
+                .filter(|function| function.name == "main")
```

Comment:
I understand that `InvalidData` will be returned in the (very unlikely) case of the script not containing `main()`. Having said that, it might be _much_ easier to debug (both for us and for a Sway user completely unaware of the internals of the SDK) if we check, right here, whether we found `main` or not, and pop up a very explicit error if that's not the case.